### PR TITLE
feat(list): adding list item selection sample

### PR DESCRIPTION
**Pull request by kdinev**
Adding a new sample showing how to implement selected styles to list items.

### CODE REVIEW

File: src/app/lists/list/list-item-selection/list-item-selection.component.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        <igx-list-item isHeader="true">Contacts</igx-list-item>
+        <igx-list-item igxRipple
+                       igxRippleTarget=".igx-list__item-content"
+                       [ngClass]="contact.selected ? 'selected' : ''"
+                       (click)="selectItem(contact)"
+                       *ngFor="let contact of contacts | igxFilter: filterContacts;">
+            <igx-avatar igxListThumbnail [src]="contact.photo" roundShape="true"></igx-avatar>
+            <span igxListLineTitle>{{ contact.name }}</span>
+            <span igxListLineSubTitle>{{ contact.phone }}</span>
+            <igx-icon igxListAction [style.color]="contact.isFavorite ? 'orange' : 'lightgray'" (click)="toggleFavorite(contact)">star</igx-icon>
```
Comment (wnvko) [inline src/app/lists/list/list-item-selection/list-item-selection.component.html:22]: clicking on the favorite start will also select the list item. IMO this is not ok. You can pass event to `toggleFavoriet` and call `event.stopPropagation` .

File: src/app/lists/list/list-item-selection/list-item-selection.component.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      {
+        isFavorite: true,
+        name: 'Dorothy H. Spencer',
+        phone: '573-394-9254',
+        photo: 'assets/images/women/67.jpg',
+        selected: false
+      }
+    ];
+
+    public displayDensities;
```
Comment (wnvko) [inline src/app/lists/list/list-item-selection/list-item-selection.component.ts:50]: is this leftover?

**Review by wnvko — CHANGES_REQUESTED** [review]
There is no config added for this sample in `live-editing/configs/ListConfigGenerator.ts`

**Review by wnvko — DISMISSED** [review]

**Review by wnvko — DISMISSED** [review]

File: live-editing/configs/ListConfigGenerator.ts [hunk 3]
```diff
@@ -133,6 +133,19 @@ export class ListConfigGenerator implements IConfigGenerator {
             shortenComponentPathBy: "/lists/list/"
         }));
 
+        // list item selection
+        configs.push(new Config({
+            component: 'ListItemSelectionComponent',
+            appModuleConfig: new AppModuleConfig({
+                imports: ['IgxAvatarModule', 'IgxFilterModule', 'IgxIconModule', 'IgxListModule',
+                    'IgxInputGroupModule', 'IgxRippleModule', 'ListItemSelectionComponent'],
```
Comment (wnvko) [inline live-editing/configs/ListConfigGenerator.ts:141]: `IgxRippleModule` is not needed anymore here and in `ngImports`

File: src/app/lists/list/list-item-selection/list-item-selection.component.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      {
+        isFavorite: true,
+        name: 'Dorothy H. Spencer',
+        phone: '573-394-9254',
+        photo: 'assets/images/women/67.jpg',
+        selected: false
+      }
+    ];
+
+    constructor() { }
```
Comment (wnvko) [inline src/app/lists/list/list-item-selection/list-item-selection.component.ts:50]: empty constructor could be deleted.

**Review by wnvko — CHANGES_REQUESTED** [review]

File: live-editing/configs/ListConfigGenerator.ts [hunk 5]
```diff
@@ -133,6 +133,19 @@ export class ListConfigGenerator implements IConfigGenerator {
...
 
+        // list item selection
+        configs.push(new Config({
+            component: 'ListItemSelectionComponent',
+            appModuleConfig: new AppModuleConfig({
+                imports: ['IgxAvatarModule', 'IgxFilterModule', 'IgxIconModule', 'IgxListModule',
+                    'IgxInputGroupModule', 'ListItemSelectionComponent'],
+                ngDeclarations: ['ListItemSelectionComponent'],
+                ngImports: ['IgxAvatarModule', 'IgxFilterModule', 'IgxIconModule', 'IgxListModule',
+                    'IgxInputGroupModule', 'IgxRippleModule']
```
Comment (wnvko) [inline live-editing/configs/ListConfigGenerator.ts:144]:
```suggestion
                    'IgxInputGroupModule']
```

**Review by wnvko — DISMISSED** [review]

**Comment by kdinev** [comment]
@wnvko Have you also tested it?

**Review by wnvko — APPROVED** [review]

**Pull request merged**